src/OpenTelemetry.Api/Context/Propagation/CompositeTextMapPropagator.cs

```diff
@@ -11,8 +11,8 @@ namespace OpenTelemetry.Context.Propagation;
 /// </summary>
 public class CompositeTextMapPropagator : TextMapPropagator
 {
-    private static readonly ISet<string> EmptyFields = new HashSet<string>();
-    private readonly List<TextMapPropagator> propagators;
+    private readonly IReadOnlyList<TextMapPropagator> propagators;
```

Comment:
Below in `Extract` and `Inject` we perform...

```csharp
foreach (var propagator in this.propagators)
{
   ...
}
```

IIRC switching this from `List<TextMapPropagator>` to `IReadOnlyList<TextMapPropagator>` will cause the `struct` `List<T>.Enumerator` to be boxed into `IEnumerator` which causes an allocation.

Let's go back to `List<TextMapPropagator>` or if you _really_ want to use `IReadOnlyList<TextMapPropagator>` you could switch the loops to do something like...

```csharp
for (int i = 0; i < this.propagtors.Count; i++)
{
   var propagator = this.propagators[i];
   ...
}
```

Reply:
Thanks for catching this @CodeBlanch. Indeed, it was allocating 40B. I have switched to `for` loops, as suggested.